src/js/game/hud/parts/mass_selector.js

```diff
@@ -155,13 +148,12 @@ export class HUDMassSelector extends BaseHUDPart {
     }
 
     clearBelts() {
-        for (const uid of this.selectedUids) {
-            const entity = this.root.entityMgr.findByUid(uid);
+        for (const entity of this.selectedEntities) {
             for (const component of Object.values(entity.components)) {
                 /** @type {Component} */ (component).clear();
             }
         }
-        this.selectedUids = new Set();
+        this.selectedEntities = [];

```

Comment:
This is the only place selectedUid's were used so it feels like it might be leftover from my branches changes. Moved everything to the selectedEntities array and it seems to be working flawlessly